### PR TITLE
feat: more aggressive autoclose on .gitignore

### PR DESCRIPTION
**Pull request by naomi-lgbt**
Checklist:

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

Changes the `.gitignore` logic to auto-close any PR to that file (if it is the only file) unless it comes from a member of the mod team.

### CODE REVIEW

**Comment by naomi-lgbt** [comment]
Do we want to remove the automatic spam label with this change?

**Review by raisedadead — COMMENTED** [review]
I was wondering if the check could be simpler:

1. One file being changed - `.gitignore`
2. They are a first timer

**Review by ojeytonwilliams — DISMISSED** [review]
LGTM :+1: 

I'm okay with the spam label staying.  Chances of false positives seem pretty slim, given that it's rare to want to modify just the .gitignore file.

**Comment by raisedadead** [comment]
> 1. One file being changed - `.gitignore`
> 2. They are a first timer

To expand, I am pretty sure these two conditions will cover all spam to that particular files. A legit contributor will rarely make changes to the file, and a pior contributor will have reasons to make changes.

**Comment by naomi-lgbt** [comment]
> I was wondering if the check could be simpler
>
> One file being changed - .gitignore
> They are a first timer

The logic isn't much simpler, but I think this is a safer approach to avoid false positives.

**Review by raisedadead — DISMISSED** [review]

File: .github/workflows/autoclose.yml [hunk 1]
```diff
@@ -14,38 +14,45 @@ jobs:
...
+
+            for (const issue of issues) {
+              if (issue.number === context.issue.number) {
+                continue;
+              }
+
+              if (issue.pull_request) {
+                return // Creator is already a contributor.
+              }
+            }
```
Comment (ojeytonwilliams) [inline .github/workflows/autoclose.yml:39]: The logic here is "if have PR and not this one, then must have contributed"?  Could we add a comment to that effect?  I spent too long trying to understand it.

**Review by ojeytonwilliams — APPROVED** [review]
Thanks!

**Review by raisedadead — APPROVED** [review]

**Comment by gikf** [comment]
Workflow errored https://github.com/freeCodeCamp/freeCodeCamp/actions/runs/3204624389/jobs/5236137041

**Comment by ojeytonwilliams** [comment]
I blame the reviewers...

**Pull request merged**